README.md

```diff
@@ -107,3 +110,64 @@ To list all entitties installed
 ```shell
 quarkus backstage entities list
 ```
+
+### Templates
+
+#### Generating a Backstage Template
+
+The backstage extension is able to generate a backstage template from an existing Quarkus application.
+The generated template will include a parmeterized version of the project and a template definition.
+
+##### Generated Template
+
+###### Parameters
+The following parameters are generated:
+- `componentId`: The component id that will be used in the backstage catalog
+- `groupId`: The group id of the project
+- `artifactId`: The artifact id of the project
+- `version`: The version of the project
+- `description`: The description of the project
+- `name`: The name of the project
+- `package`: The base package of the project
+###### Steps
+
+- `render`: Render the project template.
+- `publish`: Publish the generated code to github.
+- `register`: Register the component in the backstage catalog.
+
+
+#### Generating a Template using the CLI
+
+To generate a backstage template from an existing Quarkus application:
+
+```shell
+quarkus backstage template generate
+```
+
+#### Listing Backstage Templates
+
+To list all the backstage templates:
+
+```shell
+quarkus backstage template list
+```
+
+The command will generate a template under the `.backstage/templates` directory.
+The template can then be manually imported to backstage.
+
+
+#### Installing a Backstage Template
+
+The generated template can be installed to backstage using the following command:
```

Comment:
Has the user using the Quarkus CLI or extension the same RBAC as users accessing the backstage UI ?

Reply:
Yes